### ISSUE TITLE
Globally activated packages don't respect the sdk they were activated with

### ISSUE THREAD

**Issue by jakemac53** [dart-lang/pub#2128]
This has been coming up in the context of `flutter_web`, which requires a globally activated package (`webdev`).

If you globally activate it using the flutter SDK - like `flutter packages global activate webdev`, but you have the regular dart sdk on your PATH, then it will end up using the Dart SDK and not the Flutter SDK to run webdev (if you use `webdev` from your path).

This is because the webdev executable (a shell script) runs the snapshot like this `dart <snapshot-path>`.

We should consider if it would make sense for pub to inject an absolute path to the `dart` binary from the sdk that was used to globally activate the package, instead of relying on `dart` from your PATH.

**Comment by DanTup** [comment]
It might also be nice to have some error handling in the script so that if it fails (because `dart` wasn't found) it gives a more useful message about what might have gone wrong and how to fix.

**Comment by timsneath** [comment]
We should figure out what to tell people in the [instructions](https://github.com/flutter/flutter_web/) -- at present, we don't tell them to put the Dart SDK in their path.

**Comment by DanTup** [comment]
Yeah, it also feels slightly weird to tell Flutter users to put `flutter/bin/cache/dart-sdk/bin` in their PATH. It's not required for the IDE integrations, but there seem to be a lot of people using the cli not realising they could use IDEs (maybe we should make that clearer too, including the support for restart-on-save).

**Comment by sigurdm** [comment]
This is very old. I think it is somewhat obsolete. Now `$FLUTTER_ROOT/bin/dart` exists, and is - I guess - the right way to run and activate global packages...

**Comment by sigurdm** [comment]
In the future I think we would like to precompile executables from globally activated packages, such that they don't depend on the installed sdk at all.
